pkgs/games/tlauncher/default.nix

```diff
@@ -0,0 +1,89 @@
+{ lib
+, stdenvNoCC
+, openjdk8
+, buildFHSUserEnv
+, fetchzip
+, copyDesktopItems
+, makeDesktopItem
+}:
+let
+  version = "2.839";
+  src = stdenvNoCC.mkDerivation {
+    name = "tlauncher-${version}.jar";
+    src = fetchzip {
+      name = "tlauncher.zip";
+      url = "https://dl2.tlauncher.org/f.php?f=files%2FTLauncher-${version}.zip";
+      sha256 = "sha256-KphpNuTucpuJhXspKxqDyYQN6vbpY0XCB3GAd5YCGbc=";
+      stripRoot = false;
+    };
+    installPhase = ''
+      cp $src/*.jar $out
+    '';
+  };
+  fhs = buildFHSUserEnv {
+    name = "tlauncher";
+    runScript = ''
+      ${openjdk8}/bin/java -jar "${src}" "$@"
+    '';
+    targetPkgs = pkgs: with pkgs; [
+      alsa-lib
+      cpio
+      cups
+      file
+      fontconfig
+      freetype
+      giflib
+      glib
+      gnome2.GConf
+      gnome2.gnome_vfs
+      gtk2
+      libjpeg
+      libGL
+      openjdk8-bootstrap
+      perl
+      which
+      xorg.libICE
+      xorg.libX11
+      xorg.libXcursor
+      xorg.libXext
+      xorg.libXi
+      xorg.libXinerama
+      xorg.libXrandr
+      xorg.xrandr
+      xorg.libXrender
+      xorg.libXt
+      xorg.libXtst
+      xorg.libXtst
+      xorg.libXxf86vm
+      zip
+      zlib
+    ];
+  };
+  desktopItem = makeDesktopItem {
+    name = "tlauncher";
+    exec = "tlauncher";
+    # icon = "tlauncher"; # TODO: add icon
+    comment = "Minecraft launcher";
+    desktopName = "TLauncher";
+    categories = "Game;";
+  };
+in stdenvNoCC.mkDerivation {
+  pname = "tlauncher";
+  inherit version;
+
+  dontUnpack = true;
+
+  installPhase = ''
+    mkdir $out/{bin,share/applications} -p
+    cp ${fhs}/bin/tlauncher $out/bin
+    cp ${desktopItem}/share/applications/* $out/share/applications
```

Comment:
Please use copyDesktopItems.

Reply:
for some reason it's not working

i will do a more deep look later

happy new year